Dockerfile

```diff
@@ -0,0 +1,29 @@
+# Install Poetry and create venv in the builder step,
+# then copy the venv to the runtime image, so that the runtime image is as small as possible.
+FROM --platform=linux/amd64 python:3.10.14-slim-bookworm AS builder
+
+RUN pip install poetry==1.8.2
+
+ENV POETRY_NO_INTERACTION=1 \
+    POETRY_VIRTUALENVS_IN_PROJECT=1 \
+    POETRY_VIRTUALENVS_CREATE=1 \
+    POETRY_CACHE_DIR=/tmp/poetry_cache
+
+WORKDIR /app
+
+COPY pyproject.toml poetry.lock ./
+
+RUN --mount=type=cache,target=$POETRY_CACHE_DIR poetry install --no-root --only main
+
+FROM --platform=linux/amd64 python:3.10.14-slim-bookworm AS runtime
+
+ENV VIRTUAL_ENV=/app/.venv \
+    PATH="/app/.venv/bin:$PATH"
+
+WORKDIR /app
+
+COPY --from=builder ${VIRTUAL_ENV} ${VIRTUAL_ENV}
+
+COPY prediction_market_agent ./prediction_market_agent
+
+ENV PYTHONPATH=/app
```

Comment:
Do we need a `CMD`/`ENTRYPOINT` here or is that done in the Kubernetes config? I guess if we're building this in CI then we want to test running it too?

Reply:
Good question, command is easily changeable and I put it into Terraform https://github.com/gnosis/gnosisai-infrastructure/pull/3/files#diff-fa25fb4ef6fa67ec2382f07d26b9039099ce7a314a7e98517370a8439a0d0365R42

But if I put it here, it's easier to change it without the need for updating Terraform 🤔 I'll give it a try!

Reply:
Yep works fine, updated 🙏